src/main/java/net/mcreator/ui/modgui/ModElementGUI.java

```diff
@@ -106,15 +109,22 @@ public void setModElementCreatedListener(ModElementCreatedListener<GE> modElemen
 
 		// reload data lists in a background thread
 		this.tabIn.setTabShownListener(tab -> {
-			if (PreferencesManager.PREFERENCES.ui.autoreloadTabs)
+			if (PreferencesManager.PREFERENCES.ui.autoreloadTabs) {
+				listeningEnabled = false;
 				reloadDataLists();
+				listeningEnabled = true;
+			}
 		});
-		this.tabIn.setTabClosingListener(tab -> {
-			if (changed)
-				return 0 == JOptionPane.showConfirmDialog(mcreator, L10N.label("dialog.unsaved_changes.message"),
-						L10N.t("dialog.unsaved_changes.title"), JOptionPane.YES_NO_OPTION);
-			return true;
-		});
+
+		if (PreferencesManager.PREFERENCES.ui.remindOfUnsavedChanges) {
+			this.tabIn.setTabClosingListener(tab -> {
+				if (changed)
+					return JOptionPane.YES_OPTION == JOptionPane.showConfirmDialog(mcreator,
+							L10N.label("dialog.unsaved_changes.message"), L10N.t("dialog.unsaved_changes.title"),
+							JOptionPane.YES_NO_OPTION);
```

Comment:
If an end user makes changes to a mod element, _then_ disables the new preference and _then_ intends to close this element, they would still get the warning, and this behavior does not match the current description of that preference. I plan to correct this in a couple of hours.
No need to unassign the PR as I'll be fast in this.

Reply:
Ok

Reply:
Ready?

Reply:
Just in time :D